pkg/pillar/devicenetwork/dnc.go

```diff
@@ -594,21 +593,50 @@ func HandleAssignableAdaptersDelete(ctxArg interface{}, key string,
 	log.Infof("HandleAssignableAdaptersDelete done for %s\n", key)
 }
 
-// IngestPortConfigList creates and republishes the inintial list
-func IngestPortConfigList(ctx *DeviceNetworkContext) {
+// IngestPortConfigList creates and republishes the initial list
+// Removes bad ones
+// Optionally removes old legacy ones
+func IngestPortConfigList(ctx *DeviceNetworkContext, removeLegacy bool) {
 	log.Infof("IngestPortConfigList")
 	item, err := ctx.PubDevicePortConfigList.Get("global")
-	var dpcl types.DevicePortConfigList
+	var storedDpcl types.DevicePortConfigList
 	if err != nil {
 		log.Errorf("No global key for DevicePortConfigList")
-		dpcl = types.DevicePortConfigList{}
+		storedDpcl = types.DevicePortConfigList{}
 	} else {
-		dpcl = item.(types.DevicePortConfigList)
+		storedDpcl = item.(types.DevicePortConfigList)
+	}
+	log.Infof("Initial DPCL %v", storedDpcl)
+	var dpcl types.DevicePortConfigList
+	var hardwareModel string
+	if removeLegacy {
+		hardwareModel = hardware.GetHardwareModel()
+	}
+	for _, portConfig := range storedDpcl.PortConfigList {
+		if portConfig.CountMgmtPorts() == 0 {
+			log.Warnf("Stored DevicePortConfig key %s has no management ports; ignored",
+				portConfig.Key)
+			continue
+		}
```

Comment:
May be add another check like portConfig.HasNoValidDhcpPorts? Idea is to ignore DPC with none of it's ports having a valid DHCP type. Might not be required with your fixes in the the other parts of system, but can help when we miss a device that has this invalid configuration.

Reply:
Dhcp=0 is valid for an app shared port which is intended to be used for a switch network instance (I almost forgot about that case myself).
Thus we shouldn't look at IP addresses

Reply:
This is what I also wanted to say. Such interface's will be of not much use in NIM's context. We should skip DPCs that are full of such interfaces.